sdk/azcore/policy_retry.go

```diff
@@ -192,21 +198,27 @@ func (p *retryPolicy) Do(ctx context.Context, req *Request) (resp *Response, err
 		// drain before retrying so nothing is leaked
 		resp.Drain()
 
+		if try == p.options.MaxTries {
+			// max number of tries has been reached, don't sleep again
+			return
+		}
+
 		// use the delay from retry-after if available
 		delay, ok := resp.RetryAfter()
 		if !ok {
 			delay = p.options.calcDelay(try)
```

Comment:
By the way, we should only support exponential retries and nothing else so no need to have a retry option for the retry type. Also, we probably want a minimum delay; that is, a delay of 0 is never appropriate. Let's make a const for minimum delay and set it initially to 100 milliseconds.

Reply:
For ARM we have to honor the `Retry-After` header if available (it takes precedence over our own delay calculation).